frontend/lib/search/filter_bar_button.dart

```diff
@@ -0,0 +1,57 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/svg.dart';
+
+import '../category_assets.dart';
+
+class FilterBarButton extends StatefulWidget {
+  final int index;
+  final Function(int, bool) onCategoryPress;
+
+  FilterBarButton({Key key, this.index, this.onCategoryPress}) : super(key: key);
+
+  @override
+  State<StatefulWidget> createState() {
+    return _FilterBarButtonState(index: this.index, onCategoryPress: this.onCategoryPress);
+  }
+}
+
+class _FilterBarButtonState extends State<FilterBarButton> {
+  final int index;
+  bool _selected = false;
+  final Function(int, bool) onCategoryPress;
```

Comment:
```suggestion
  bool _selected = false;
```
You can use the widget fields like `widget.index` and `widget.onCategoryPress`